src/ServiceInsight/Framework/Modules/CoreModule.cs

```diff
@@ -21,6 +21,7 @@ protected override void Load(ContainerBuilder builder)
             builder.RegisterType<ServiceControlConnectionProvider>().InstancePerLifetimeScope();
             builder.RegisterType<DefaultServiceControl>().As<IServiceControl>().InstancePerLifetimeScope();
             builder.RegisterType<CommandLineArgParser>().SingleInstance().OnActivating(e => e.Instance.Parse());
+            builder.RegisterType<WorkNotifier>().As<IWorkNotifier>();
```

Comment:
This can be registered as Singleton, since the WorkNotifier itself is not disposable (it generates disposable objects).